mentoring/tests/integration/test_titles.py

```diff
@@ -0,0 +1,126 @@
+        ("",             None,  "Question"),
+        ("",             True,  "Question"),
+        ("",             False, None),
+    )
+
+    mcq_template = """
+        <problem-builder mode="{{mode}}">
+            <pb-mcq name="mcq_1_1" question="Who was your favorite character?"
+              correct_choices="gaius,adama,starbuck,roslin,six,lee"
+              {display_name_attr} {show_title_attr}
+            >
+                <pb-choice value="gaius">Gaius Baltar</pb-choice>
+                <pb-choice value="adama">Admiral William Adama</pb-choice>
+                <pb-choice value="starbuck">Starbuck</pb-choice>
+                <pb-choice value="roslin">Laura Roslin</pb-choice>
+                <pb-choice value="six">Number Six</pb-choice>
+                <pb-choice value="lee">Lee Adama</pb-choice>
+            </pb-mcq>
+        </problem-builder>
+    """
+
+    mrq_template = """
+        <problem-builder mode="{{mode}}">
+            <pb-mrq name="mrq_1_1" question="What makes a great MRQ?"
+              ignored_choices="1,2,3"
+              {display_name_attr} {show_title_attr}
+            >
+                <pb-choice value="1">Lots of choices</pb-choice>
+                <pb-choice value="2">Funny choices</pb-choice>
+                <pb-choice value="3">Not sure</pb-choice>
+            </pb-mrq>
+        </problem-builder>
+    """
+
+    rating_template = """
+        <problem-builder mode="{{mode}}">
+            <pb-rating name="rating_1_1" question="How do you rate Battlestar Galactica?"
+              correct_choices="5,6"
+              {display_name_attr} {show_title_attr}
+            >
+                <pb-choice value="6">More than 5 stars</pb-choice>
+            </pb-rating>
+        </problem-builder>
+    """
+
+    long_answer_template = """
+        <problem-builder mode="{{mode}}">
+            <pb-answer name="answer_1_1" question="What did you think of the ending?"
+              {display_name_attr} {show_title_attr} />
+        </problem-builder>
+    """
+
+    def setUp(self):
+        super(StepTitlesTest, self).setUp()
+        # Disable asides for this test since the acid aside seems to cause Database errors
+        # When we test multiple scenarios in one test method.
+        patcher = patch(
+            'workbench.runtime.WorkbenchRuntime.applicable_aside_types',
+            lambda self, block: [], create=True
+        )
+        patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def test_all_the_things(self):
+        """ Test various permutations of our problem-builder components and title options. """
+        # We use a loop within the test rather than DDT, because this is WAY faster
+        # since we can bypass the Selenium set-up and teardown
+        for display_name, show_title, expected_title in self.test_parameters:
+            for mode in ("standard", "assessment"):
+                for qtype in ("mcq", "mrq", "rating", "long_answer"):
```

Comment:
@bradenmacdonald nit: this could be a parameterized test. I believe you should still recall why I prefer multiple parameterized tests over single monolithic with loops in it. :)